code/game/gamemodes/objective.dm

```diff
@@ -290,47 +308,39 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 	needs_target = FALSE
 
 /datum/objective/block/check_completion()
-	if(!istype(owner.current, /mob/living/silicon))
-		return FALSE
+	for(var/datum/mind/M in get_owners())
+		if(!M.current || issilicon(M.current))
+			return FALSE
 	if(SSticker.mode.station_was_nuked)
 		return TRUE
 	if(SSshuttle.emergency.mode < SHUTTLE_ENDGAME)
 		return FALSE
-	if(!owner.current)
-		return FALSE
 	if(!SSshuttle.emergency.is_hijacked(TRUE))
 		return FALSE
-
 	return TRUE
 
 /datum/objective/escape
 	explanation_text = "Escape on the shuttle or an escape pod alive and free."
 	needs_target = FALSE
 
 /datum/objective/escape/check_completion()
-	if(issilicon(owner.current))
-		return 0
-	if(isbrain(owner.current))
-		return 0
-	if(!owner.current || owner.current.stat == DEAD)
-		return 0
+	if(SSshuttle.emergency.mode < SHUTTLE_ENDGAME)
+		return FALSE
 	if(SSticker.force_ending) //This one isn't their fault, so lets just assume good faith
-		return 1
+		return TRUE
 	if(SSticker.mode.station_was_nuked) //If they escaped the blast somehow, let them win
-		return 1
-	if(SSshuttle.emergency.mode < SHUTTLE_ENDGAME)
-		return 0
-	var/turf/location = get_turf(owner.current)
-	if(!location)
-		return 0
-
-	if(istype(location, /turf/simulated/floor/mineral/plastitanium/red/brig)) // Fails traitors if they are in the shuttle brig -- Polymorph
-		return 0
-
-	if(location.onCentcom() || location.onSyndieBase())
-		return 1
+		return TRUE
+	var/list/owners = get_owners()
+	for(var/datum/mind/M in owners)
+		if(M.current?.stat == DEAD || issilicon(M.current) || isbrain(M.current))
```

Comment:
If there is no `M.current` this if won't evaluate and will result in an eventual return of TRUE in certain cases. That seems unintentional.

Reply:
You're right, I'll need to still check `!M.current || M.current.stat == DEAD`